lib/correios/cep/web_service.rb

```diff
@@ -14,15 +14,11 @@ def initialize
       def request(zipcode)
         http = build_http
 
-        request = build_request(zipcode)
-        Correios::CEP.log_request(request, uri.to_s)
-
-        response = http.request(request)
-        Correios::CEP.log_response(response)
-
-        http.finish if http.started?
+        response = request_with_retry { http.request(build_request(zipcode)) }
```

Comment:
I believe `EOFError` only happens when we run the request, right?
Therefore building the request can stay outside of `request_with_retry` block.